linkerd/proxy/detect/src/lib.rs

```diff
@@ -64,7 +60,7 @@ impl<D, A> Accept<D, A> {
 
 impl<T, D, A> tower::Service<(T, BoxedIo)> for Accept<D, A>
 where
-    D: Detect<T>,
+    D: Detect<T> + Clone,
```

Comment:
From a local build/test, it doesn't seem this `Clone` is necessary. Any reason it was added?